.github/workflows/promptflow-release-testing-matrix.yml

```diff
@@ -13,28 +13,58 @@ on:
         required: false
         type: string
 env:
-  packageSetupType: promptflow_with_extra
   testWorkingDirectory: src/promptflow
   PYTHONPATH: ${{ github.workspace }}/src/promptflow
   IS_IN_CI_PIPELINE: "true"
 jobs:
-  id:
+  build:
     runs-on: ubuntu-latest
     steps:
-      - name: workflow run id - ${{ inputs.id }}
-        run: |
-          echo "workflow run id: ${{ inputs.id }}"
-  all_tests:
+    - name: workflow run id - ${{ inputs.id }}
+      run: |
+        echo "workflow run id: ${{ inputs.id }}"
+    - name: checkout
+      uses: actions/checkout@v4
+    - name: Python Setup - 3.9
+      uses: "./.github/actions/step_create_python_environment"
+      with:
+        pythonVersion: 3.9
+    - name: Build wheel
+      uses: "./.github/actions/step_sdk_setup"
+      with:
+        setupType: promptflow_with_extra
+        scriptPath: ${{ env.testWorkingDirectory }}
+    - name: Upload Wheel
+      uses: actions/upload-artifact@v3
+      with:
+        name: wheel
+        path: |
+          ${{ github.workspace }}/src/promptflow/dist/*.whl
+          ${{ github.workspace }}/src/promptflow-tools/dist/*.whl
+  promptflow_sdk_cli_tests:
+    if: ${{ github.event_name == 'workflow_dispatch' || github.event_name == 'workflow_call' || github.event_name == 'pull_request' }}
+    needs: build
+    env:
+      PROMPT_FLOW_TEST_MODE: "live"
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-latest, macos-latest, windows-latest]
-        testType: [sdk-cli, executor]
+        os: [ubuntu-latest, windows-latest, macos-latest]
         pythonVersion: ['3.8', '3.9', '3.10', '3.11']
     runs-on: ${{ matrix.os }}
     steps:
+    - name: set test mode
+      shell: pwsh
+      run: |
+        # Windows support of multiprocessing is different from other platforms, mock injections is not working properly in replay mode.
+        # Mac is not stable in replaying test.
+        if (("${{ matrix.os }}" -eq "windows-latest") -or ("${{ matrix.os }}" -eq "macos-latest")) {
+          echo "PROMPT_FLOW_TEST_MODE=live"  | Out-File -FilePath $Env:GITHUB_ENV -Encoding utf8 -Append
+        } else {
+          echo "PROMPT_FLOW_TEST_MODE=replay"  | Out-File -FilePath $Env:GITHUB_ENV -Encoding utf8 -Append
+        }
     - name: checkout
-      uses: actions/checkout@v4
+      uses: actions/checkout@v3
```

Comment:
we have inconsistent version for `actions/checkout` in this file, let's align them

Reply:
Good catch